### PR TITLE
Rework types.

### PR DESCRIPTION
**Pull request by mshima**
- rename generator-base.mjs to generator-base-todo.mjs (apis should be refactored, converted to typescript and moved to others generators)
- rename generator-base-private.mjs to generator-base-definition.mjs (changes yeoman-generator apis. It's not possible to change some typescript apis, change using js and jsdoc)
- convert generator-base-blueprint to typescript
- rename base/generator.mts to base/generator-base.mts (blueprint support is quite big, make generator-base.mts a generator-blueprint parent and don't integrate the blueprint support in generator-base).
- rework priorities, tasks and tasksGroups types.
- convert base-application/generator to typescript and integrate generator-ts

Tasks types looks to be behaving a lot better now.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/base-application/generator.mts [hunk 1]
```diff
@@ -52,16 +59,22 @@ const {
   POST_WRITING_ENTITIES_QUEUE,
 } = QUEUES;
 
-const asPriority = BaseApplicationTsGenerator.asPriority;
+const asPriority = BaseGenerator.asPriority;
+
+type ApplicationDefinition = {
```
Comment (Tcharl) [inline generators/base-application/generator.mts:64]: A bit of doc? what is it made for?

File: generators/base-application/tasks.d.mts [hunk 2]
```diff
@@ -1,31 +1,36 @@
...
-  LoadingTaskGroup as BaseLoadingTaskGroup,
-} from '../base/tasks.mjs';
+import { ControlTaskParam, BaseGeneratorDefinition, SourceTaskParam } from '../base/tasks.mjs';
+import { BaseApplication } from './types.mjs';
 
-export type Application = Record<string, any>;
+type Field = {
+  fieldName: string;
+  fieldType: string;
+  fieldTypeBlobContent: string;
```
Comment (Tcharl) [inline generators/base-application/tasks.d.mts:7]: Usage of decorator pattern? BlobType is not always filled

File: generators/base-application/tasks.d.mts [hunk 3]
```diff
@@ -1,31 +1,36 @@
...
-type ConfiguringEachEntityTaskParam<ApplicationType> = ApplicationTaskParam<ApplicationType> & {
+export type Entity = {
+  fields: Field[];
+  relationships: Relationship[];
+} & Record<string, any>;
+
+type ApplicationDefinition = {
+  applicationType: BaseApplication;
+  entityType: Entity;
+  sourceType: Record<string, (...args: any[]) => void>;
```
Comment (Tcharl) [inline generators/base-application/tasks.d.mts:22]: A bit of doc please, can' t understand the entire set of attribute

File: generators/client/index.mts [hunk 4]
```diff
@@ -16,5 +16,16 @@
...
+import { BaseApplicationGeneratorDefinition, Entity } from '../base-application/tasks.mjs';
+import { ClientApplication } from './types.mjs';
+
 export { default } from './generator.mjs';
 export { files as commonFiles } from './files-common.mjs';
+
+type ApplicationDefinition = {
+  applicationType: ClientApplication;
+  entityType: Entity;
+  sourceType: Record<string, (...args: any[]) => void>;
```
Comment (Tcharl) [inline generators/client/index.mts:28]: Doc please :-)

File: generators/common/index.mts [hunk 5]
```diff
@@ -16,5 +16,19 @@
...
+import { ClientServerApplication } from './types.mjs';
+
 export { default } from './generator.mjs';
 export { commonFiles as files } from './files.mjs';
+
+// TODO move to ./generator.mts
+type ApplicationDefinition = {
+  applicationType: ClientServerApplication;
+  entityType: Entity;
+  sourceType: Record<string, (...args: any[]) => void>;
```
Comment (Tcharl) [inline generators/common/index.mts:29]: Same here, doc please

File: generators/cypress/generator.mts [hunk 6]
```diff
@@ -27,24 +27,20 @@ import { CLIENT_MAIN_SRC_DIR } from '../generator-constants.mjs';
...
 
-/**
- * @class
- * @extends {BaseApplicationGenerator<CypressApplication>}
- */
-export default class CypressGenerator extends BaseApplicationGenerator<CypressApplication> {
+type ApplicationDefinition = {
+  applicationType: CypressApplication;
+  entityType: Entity;
+  sourceType: Record<string, (...args: any[]) => void>;
```
Comment (Tcharl) [inline generators/cypress/generator.mts:38]: Doc

Comment (Tcharl) [reply generators/cypress/generator.mts:38]: May this types extracted out of the generator.mts? In a dedicated file like `generator-types.d`?

Would be nice to slim down the generator LOCs

**Comment by Tcharl** [comment]
Wow, this is huge, great!
May you please add a bit of doc on types and extract them on specific -type.d files?
The smallest the generator.mts files will be, the best the developper experience will be!

**Comment by mshima** [comment]
Types are not finalized yet, I have a follow-up PR with entities.
Docs can be added later.
Generator types are related to the generator, it’s in the same file because type imports can be reused. And it’s available at a single glance.

File: generators/base-application/tasks.d.mts (hunk 2 above)
Comment (mshima) [reply generators/base-application/tasks.d.mts:7]: This was moved from below.
Fields types will be worked on later

File: generators/cypress/generator.mts (hunk 6 above)
Comment (mshima) [reply generators/cypress/generator.mts:38]: IMO this is much better here.
It’s available at a glance where it’s used.

**Comment by mshima** [comment]
Jsdoc will be added later, some utilities types can be added and those jsdoc will need rework too.
And avoids conflicts with the follow-up PR.
Not worth to add jsdoc right now.

File: generators/cypress/generator.mts (hunk 6 above)
Comment (Tcharl) [reply generators/cypress/generator.mts:38]: I was taking my good old OOP ayatolla posture ^^. `type CypressApplicationDefinition extends ApplicationDefinition<CypressApplication>`

Comment (Tcharl) [reply generators/cypress/generator.mts:38]: I was taking my good old OOP ayatolla posture ^^. `type CypressApplicationDefinition extends ApplicationDefinition<CypressApplication>`

**Pull request merged**